lib/Dialect/AIE/IR/AIEDialect.cpp

```diff
@@ -1545,60 +1547,93 @@ LogicalResult DMABDOp::verify() {
+           "expected MemTileDMAOp parent");
+    auto memOp = getOperation()->getParentOfType<MemTileDMAOp>();
+    if (auto bufferOp = getBufferOp();
+        bufferOp.getTileOp().rowIndex() != memOp.rowIndex() ||
+        std::abs(bufferOp.getTileOp().colIndex() - memOp.colIndex()) > 1)
+      return emitOpError("Mem tile DMAs can only access adjacent mem tiles");
+    // TODO(max): check even channel -> <24; odd channel -> > 24
+    if (std::optional<int32_t> bdId = getBdId(); bdId.has_value() && *bdId > 47)
+      return emitOpError("Memtile DMAs have at most 48 buffer descriptors");
+    if (std::optional<int32_t> nextBdId = getNextBdId();
+        nextBdId.has_value() && *nextBdId > 47)
+      return emitOpError("Memtile DMAs have at most 48 buffer descriptors");
+    if (auto dims = getDimensions(); dims.has_value()) {
+      TileElement tile = getParentTileElement(getOperation());
+      if (getTargetModel(getOperation())
+              .isMemTile(tile.getTileID().col, tile.getTileID().row)) {
+        if (dims->size() == 4 && dims->back().getStride() != 1)
+          return emitOpError(
+              "Only stride = 1 supported for inner-most (4th) dim");
+      } else if (dims->size() == 4 && dims->back().getStride() != 1)
+        return emitOpError(
+            "Only stride = 1 supported for inner-most (4th) dim");
+    }
   }
 
   // The following checks only apply if non-default strides/wraps are defined.
   if (getDimensions()) {
-    MemRefType buffer = getBuffer().getType();
-    // We are not restrictive about the type of the memref used as the input
-    // to the DMABD when used with multi-dimensional strides/wraps. Since the
-    // BD will use the memref as a base address and copy from it in 32 bit
-    // chunks, while assuming the layout of the memref is contiguous. We
-    // assume the user/compiler understands and accounts for this.
-    uint64_t memrefSize = 1; // in bytes
-    uint64_t maxIdx = 0;
-    for (int64_t memrefDim : buffer.getShape())
-      memrefSize *= 4 * memrefDim;
-
     ArrayRef<BDDimLayoutAttr> dims = *getDimensions();
     size_t maxNDims = 3;
     if (isa_and_nonnull<MemTileDMAOp>(getOperation()->getParentOp()))
       maxNDims = 4;
-
     if (dims.size() > maxNDims)
       return emitOpError() << "Cannot give more than "
                            << std::to_string(maxNDims)
                            << " dimensions for step sizes and wraps in this "
                               " tile (got "
                            << std::to_string(dims.size()) << " dimensions).";
 
+    MemRefType buffer = getBuffer().getType();
+    int64_t maxIdx = 0;
     for (BDDimLayoutAttr dim : dims) {
       maxIdx += dim.getStride() * (dim.getSize() - 1);
       if (0 == dim.getStride())
         return emitOpError()
                << "Invalid step size; must be a positive integer.";
-      if (dim.getStride() > memrefSize)
+      if (dim.getStride() > buffer.getNumElements())
```

Comment:
These changes seem suspicious to me, since they seem to preclude having memref view transformations?   It's not clear to me why this is better, I guess.